### PR TITLE
Add a simple security disclosure policy

### LINKED ISSUE #2395: Add SECURITY.md and create a security disclosure process
https://github.com/cortexproject/cortex/security

### PR DESCRIPTION
**Pull request by gouthamve**
While extremely simple, this is lifted from the [Prometheus](https://prometheus.io/docs/operating/security/) and [k8s](https://kubernetes.io/docs/reference/issues-security/security/) security vulnerability disclosure policies.

I think this would serve our purposes for the near future and we can always adapt if this turns out to be incomplete or problematic.
 
Fixes #2395

### CODE REVIEW

**Comment by RichiH** [comment]
Could benefit from some fiddling and wordsmithing in the medium term, but good to merge as-is, IMO.

**Comment by gouthamve** [comment]
> Could benefit from some fiddling and wordsmithing in the medium term

What do you mean? Could you suggest the fiddling?

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Cortex Security and Disclosure Information
+
+As with any complex system, it is not possible to guarantee that there are no bugs. If you find a _security bug_ please report it privately to the maintainers listed in the MAINTAINERS of the relevant repository and CC cortex-team@googlegroups.com. We will fix the issue and coordinate a release date with you, acknowledging your effort and mentioning you by name if you want.
```
Comment (RichiH) [inline SECURITY.md:3]:
```suggestion
As with any complex system, it is certain that bugs will be found, some of them security-relevant.. If you find a _security bug_ please report it privately to the maintainers listed in the MAINTAINERS of the relevant repository and CC cortex-team@googlegroups.com. We will fix the issue as soon as possible and coordinate a release date with you. You will be able to choose if you want public acknowledgement of your effort and if you want to be mentioned by name.
```

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+## Public Disclosure Timing
+
+A public disclosure date is negotiated by the Cortex Team and the bug submitter. We prefer to fully disclose the bug as soon as possible once a user mitigation is available. It is reasonable to delay disclosure when the bug or the fix is not yet fully understood, the solution is not well-tested. The timeframe for disclosure is from immediate (especially if it’s already publicly known) to a few weeks. For a vulnerability with a straightforward mitigation, we expect report date to disclosure date to be on the order of 7 days. The Cortex Team holds the final say when setting a disclosure date.
```
Comment (RichiH) [inline SECURITY.md:7]:
```suggestion
The public disclosure date is agreed between the Cortex Team and the bug submitter. We prefer to fully disclose the bug as soon as possible, but only after a mitigation or fix is available. We will ask for delay if the bug or the fix is not yet fully understood or the solution is not tested to our standards yet. While there is no fixed timeframe for fix & disclosure, we will try our best to be quick and do not expect to need the usual 90 days most companies ask or. For a vulnerability with a straightforward mitigation, we expect report date to disclosure date to be on the order of 7 days.
```

File: SECURITY.md (hunk 1 above)
Comment (gouthamve) [reply SECURITY.md:3]: This is better! But the irony is that this is taken from the Prometheus docs :upside_down_face: 

Will incorporate these changes.

Comment (RichiH) [reply SECURITY.md]: So we should change those, too :)

**Pull request merged**